### DISCUSSION TITLE
Preparing for Assignment 2

### DISCUSSION THREAD

**Discussion by ebajcar** [ebajcar/web10199_worksheets#11] [Weekly activities]
1. Set up the project — [instructions here](https://www.notion.so/Developing-web-project-b3c9fab6893e4cfba6f04fb91af6c0d9)
2. Complete the analysis and design of your solution for Rock-Paper-Scissors-Lizard-Spock game — [information here](https://www.notion.so/Developing-game-of-RPSLS-3ee89452947843dd9d0662e3cc8d403d)

Ask questions here — to validate your assumptions and to obtain additional details. You can share code snippets and ideas here to help and learn from each other.

**Comment by smileliv** [comment]
@ebajcar Hi Professor Bajcar,
after following the instructions and information on the assignment, I have started working on it.
However, I found there are no resource images for the "rock", "paper", "scissors", etc
can we use our own image? or is it somewhere in your repo?
Thank you very much,
Jiali Wang

**Reply by ebajcar** [reply]
The images can be found in this repository. If you click on the "Code" above, it will take you to the root directory of this repository. where you see the README file with the links to Worksheets. 

Above the README file, you will see directory (folder) `syst10199`, click on it, then click on projects. The entire project structure is called `home2021`, with the index files and `css` and `images` sub-directories.  All images are in `images` and the css files in `css` subdirectories.

>  I just added the missing images now!

**Reply by ebajcar** [reply]
If anyone wants to use their own images, make sure all images are the same size(both the object on the canvas, and the canvas (around 120x120), thematically consistent (photo vs. drawing vs. cartoon ), same format (png), and fit in the color scheme of the page.

**Comment by ebajcar** [comment]
Hi,
just like with a css file, just click on the link in the "view code" to display the image, then you can right-click and save-as, or better to go to the github directory, they are all in the images sub-directory.
Ellen
________________________________

**Reply by smileliv** [reply]
Hi Professor Bajcar,
I saw the images with the blue pattern background for the main "member shart here" page.
The missing images are the black images in the "Game of Rock-Paper-Scissors-Lizard-Spock" page:
<img width="250" alt="Screen Shot 2021-05-25 at 12 38 19 PM" src="https://user-images.githubusercontent.com/84793649/119535600-2e816080-bd56-11eb-8442-94f52c3e822b.png">
are they in some other directory?
Thanks,
Jiali Wang

**Reply by ebajcar** [reply]
I just added the missing images now.

**Comment by evachu86** [comment]
Here is my analysis for the RPSLS game:

1. The user clicks scissors.png on the Start Here page, and the browser loads the RPSLS game page, triggering callRPSLS() function.
2. The user clicks the gesture <img> element, triggering loadResult() function.
    a. Call randomGesture() function for computer's choice after DOM ready.
    b. Display gesture by user and computer respectively.
    c. Call compareGesture() function to compare the gestures and return the result.
3. The browser loads the result page, randomly generates a gesture, and shows the comparison, there is also a "Play Again!" button.
4. Click the "Play Again!" button will trigger callRPSLS() function.

Besides, I did write some pseudo-code, but I'm not sure whether it will result in the breach of academic integrity.

**Reply by evachu86** [reply]
Part of my pseudo-code:

```
function loadRPSLS () {
    print "Welcome";
    print "Your move...click on an image to choose!";
    display rock1.png, pepar1.png, scissors1.png, lizard1.png, spock1.png;
}

// set to element rock1.png, paper1.png, scissors1.png, lizard1.png, spock1.png click event.
// rock1.png onclick = "loadResult(0)"
// paper1.png onclick = "loadResult(1)"
// scissors1.png onclick = "loadResult(2)"
// lizard1.png onclick = "loadResult(3)"
// spock1.png onclick = "loadResult(4)"
function loadResult (userGesture) {
    var resStr, computerGesture = randomGesture(),
    result = compareGesture(userGesture, computerGesture);
    
    displayGesture(computerGesture, "computerGesture");
    displayGesture(userGesture, "userGesture");
    
    if(result > 0) 
        resStr = 'YOU WIN!';
    else if(result == 0) 
        resStr = 'TIE!';
    else 
        resStr = 'MACHINE WINS!';
    
    document.getElementById("result").innerHTML = resStr;
}

function randomGesture () {
    return Math.floor(Math.random() * 4);
}

function displayGesture (gestureIndex, elementId) {
    var gesture = ["rock", "paper", "scissors", "lizard", "spock"];
    
    document.getElementById(elementId).setAttribute("src", "image/" +gesture[gestureIndex]+ "1.png");
}

function compareGesture (userGesture, computerGesture) {
    switch(userGesture) {
        0: return rockRule(computerGesture);
        1: return paperRule(computerGesture);
        2: return scissorsRule(computerGesture);
        3: return lizardRule(computerGesture);
        4: return spockRule(computerGesture);
        default:
            alert('Your gesture is not in the rule.');
            loadRPSLS();
    }
}

function rockRule (computerGesture) {
    var result = 0;
    
    switch(computerGesture) {
        1: result = -1;
            break;
        2: 
        3: result = 1;
            break;
        4: result = -1;
            break;
        default:
    }
    
    return result;
}

function paperRule (computerGesture) {
    var result = 0;
    
    switch(computerGesture) {
        0: result = 1;
            break;
        2: 
        3: result = -1;
            break;
        4: result = 1;
            break;
        default:
    }
    
    return result;
}

function scissorsRule (computerGesture) {
    var result = 0;
    
    switch(computerGesture) {
        0: result = -1;
            break;
        1: 
        3: result = 1;
            break;
        4: result = -1;
            break;
        default:
    }
    
    return result;
}

function lizardRule (computerGesture) {
    var result = 0;
    
    switch(computerGesture) {
        0: result = -1;
            break;
        1: result = 1;
            break;
        2: result = -1;
            break;
        4: result = 1;
            break;
        default:
    }
    
    return result;
}

function spockRule (computerGesture) {
    var result = 0;
    
    switch(computerGesture) {
        0: result = 1;
            break;
        1: result = -1;
            break;
        2: result = 1;
            break;
        3: result = -1;
            break;
        default:
    }
    
    return result;
}
```

**Reply by evachu86** [reply]
The analysis for RPSLS game without involvement of specific technology:
1. Start the game.
2. The user picks a signature.
3. The computer picks a signature.
4. Show the signatures from both side.
5. Compare the signature with the following rule:
![image](https://user-images.githubusercontent.com/8485777/120228014-e0e88480-c27c-11eb-8d29-4a9f3f7b03a8.png)
    O: the user wins.
    X: the user loses.
    -: fair
    reference: https://bigbangtheory.fandom.com/wiki/Rock,_Paper,_Scissors,_Lizard,_Spock
6. Declare the winner.
7. End the game.

**Reply by ebajcar** [reply]
**Some great stuff here!** 
Can step 5 be simplified or what alternative ways can we do it? Can we write it in simple sentences? Such as
ROCK smashes SCISSORS and ROCK kills LIZARD.
PAPER covers ROCK and PAPER disproves SPOCK.
...

**Reply by ebajcar** [reply]
Is there any way we can reduce the multiple SWITCH statements in a single structure????

**Comment by ebajcar** [comment]
Great start.  Keep in mind that there should be no references to any specific technology - no references to functions, DOM.  Describe it as if two people were playing just using their hands.  This is important because the focus should be on how we do things and not on the technology that will implement and hopefully improve the process.

You can share pseudo-code and code snippets in these discussions.
Ellen
________________________________

**Comment by ghost** [comment]
My analysis
1. When a user clicks on the RPSLS button(scissors) he is taken to the page where he can play the RPSLS game.
2. On the screen he is presented with a set of buttons(rock, paper, scissor, lizard, hand). To play, they must choose what gesture they want to play.
3. When the user clicks one of those buttons, the computer selects the gesture it will play. It does so with a piece of code that picks one random number ranging from 0 to 4.
4. Choices are shown.
5. Referee script decides the winner based on the gestures the user and the bot chose.
6. The game ends
7. The user is presented with an option to start again. 
8. When the user clicks "Start again" the game restarts.

**Comment by ebajcar** [comment]
### First pass algorithm for RPSLS
_Note the lack of references to any technology!_

1. Start a game
2. Human chooses one of the objects
3. Machine chooses one of the objects
4. Show (See) who chose what
5. Decide whose object beats the other
6. Declare winner
7. End the game

Next step? 
- research how to decide which object beats the other

**Reply by ghost** [reply]
Use the switch and write cases for each gesture type. So in the first switch, we have cases for all of the gestures. Now in each of these cases, we write another switch that will determine who wins. Small example:
```
switch(gesture){
  case "rock":
    switch(gesture){
        case "paper":
            console.log("You lost");
            break;
        case "scissor":
            console.log("Congrats champ, you won!");
            break;
    }
    break;
}
```

**Reply by faustocorreaa** [reply]
Good afternoon Professor Bajcar. To choose who wins, we can use a table to set each object a identify which object beats to others. The table could be this:

<table border="1">
		<tr>
			<td>ID</td>
			<td>Object</td>
			<td>BeatsTo</td>
		</tr>
		<tr>
			<td>1</td>
			<td>Rock</td>
			<td>{2,3,4}</td>
		</tr>
		<tr>
			<td>2</td>
			<td>Paper</td>
			<td>{5,1}</td>
		</tr>
		<tr>
			<td>3</td>
			<td>Scissors</td>
			<td>{2,4}</td>
		</tr>
		<tr>
			<td>4</td>
			<td>Lizard</td>
			<td>{2,5}</td>
		</tr>
		<tr>
			<td>5</td>
			<td>Spock</td>
			<td>{1,3}</td>
		</tr>
	</table>

**Reply by ebajcar** [reply]
Is there a way we can avoid using numbers?  
such as ROCK beats (SCISSORS, LIZARD)

Also check the relations between ROCK and PAPER (there should be only two objects, not three)

**Comment by xiaoyazz** [comment]
To decide which object beats the other, we need:
1. 2 players - a human and a machine
2. 5 options(rock, scissors, paper, lizard, and spock) and a scoreboard
3. When the human user enters RPSLS game, the user enters a loop. 
4. Prompt the user to make a choice among the 5 options and return an object A.
4. Call a function to make the machine choose among the rest 4 options and return another object B. 
5. Call a function to decide the relationship between object A and object B. It should be an if statement. For example, If object A === 'rock', object B ==='paper', it will return 'machine' which is the winner.
6. Call the function of the scoreboard with the winner and print out the result.
7. Ask the user to play it again or exit the game. If the user wants to play again, we go back to the top of the loop with 5 options. 
If the user chooses "exit", the loop will be ended and we go back to the home page.

**Reply by ebajcar** [reply]
Avoid vague references such as object A or object B. Focus on describing it in human words as if you were explaining it to another human being.  Assume they know nothing about computers, functions, etc.

**Reply by xiaoyazz** [reply]
Thank you, Professor!

**Comment by dovahkiin101** [comment]
I have few questions I want to clarify regarding to the assignment #2:

1. What are we supposed to do for this assignment, and is it really due this Tuesday (June 1, 2021)?
2. Is there a proper instruction for this assignment? If so, do you mind showing us the actual instruction about it? They were unclear when you mentioned about them last class.

Those are all I have in mind right now.

Thank you

**Reply by xianjun-wang** [reply]
Not due on Tuesday.  You can find the assignment 2 instruction on week 2 slides

**Reply by DieuHuynhVn** [reply]
I think that we will do, following things:  ( Go to Slate / Content -> week2 -> week2 link: https://ebajcar.github.io/web10199/slides_by_reveal/index_w2.html#/5 )

1/Set up the project  ( like create HTML, CSS, JS file in your NetBeans and upload it to GitHub)

2/Complete the analysis Rock-paper-scissors game in the Discussion board here

3/ Not sure, watch Tic tac Toe version of the ma'am on her GitHub and analysis what things should fix and improve it.

**Comment by RodneyRamsay1** [comment]
Do you want me to hand in working code of the game or a text file containing information on how we would make the game, without any references to technology/programming terminology (similar to how you did here https://github.com/ebajcar/web10199_worksheets/discussions/11#discussioncomment-787090 ) ?

**Reply by ebajcar** [reply]
Re-watch class recording.

**Reply by DieuHuynhVn** [reply]
ma'am, will we use NeatNeans or what apps to create an HTML file, CSS, JavsScript? I think we must learn more to take JavaScript, I cannot do the game 100% at this time.

**Comment by DieuHuynhVn** [comment]
Dear ma'am

This is my analysis:
- Overall, there are 2 players: computers and a human, and 5 options( Rock, Paper, Scissors, Lizard, Spock) for each people.
1/ a human clicks the "Start game" button -> user choose 1 option and machine take other ones randomly and reveals it at the same time.
2/ Comparing options:
In a tie ( 2 options are the same), the process will be repeated until a winner is found. 
Scissors cuts Paper
Paper covers Rock
Rock crushes Lizard
Lizard poisons Spock
Spock smashes Scissors
Scissors decapitate Lizard
Lizard eats Paper
Paper disproves Spock
Spock vaporizes Rock
Rock crushes Scissors 
3/ Finding out a winner.
4/Finish

**Reply by DieuHuynhVn** [reply]
Dear ma'am
my username: huthitha
Additionally, I set up the project, and here is my GitHub link: https://github.com/DieuHuynhVn/SYST10199-TicTacToe (only the  old code, no re-design from your code)

**Comment by SyedNaufil** [comment]
Hello Professor, I have completed the analysis and a rough algorithm for the RPSLS game. What exactly do you want us to submit in the Assignment 2 drop box on slate. The actual analysis ? the algorithm coded in JS ? the whole project ? Please let us know. Thank you

**Reply by ebajcar** [reply]
Re-watch class recording.

**Comment by MarjDC** [comment]
There are 5 weapons for this game. 

1. Rock 2. Paper 3. Scissor 4. Spock 5. Lizard.

A Simple Way to Remember Who Wins

Scissors cuts paper.
Paper covers rock.
Rock crushes lizard.
Lizard poisons Spock.
Spock smashes scissors.
Scissors decapitates lizard.
Lizard eats paper.
Paper disproves Spock.
Spock vaporizes rock.
Rock crushes scissors.


Number of players: 2 


Stand across from a partner
Hold your fist out in front of you and count to 3
Make one of the 5 hand signals when you say 3

Choosing what to throw:

Throw rock to crush scissors or lizard.
Use paper to cover rock or disprove Spock.
Play scissors to cut paper or decapitate lizard
Choose lizard to eat paper or poison Spock. 
Use Spock to vaporize rock or smash the scissors.

**Comment by yousef-ek80** [comment]
Analysis for RPSLS game :
RPSLS is a two player game, in this case it is human player vs computer. Each player chooses either rock, paper, scissors, Lizard or Spock. The items are compared, and whichever player chooses the more powerful item wins.

1. User starts game.
2. User picks Item.
3. Computer picks Item randomly.
4. The two items are compared.
5. The more powerful items wins and 
6. The winner is declared.
7. End of game.  

The possible outcomes are:

If you pick : 
1. rock can beat scissors and Lizard BUT rock loses to paper and spock.

2. paper can beat rock and spock  BUT  paper loses to scissors and lizard.

3. Scissors can beat paper and Lizard BUT Scissors loses to rock and spock.

4. Lizard can beat paper and spock BUT lizard loses to rock and scissors.

5. Spock  can beat scissors and rock BUT spock loses to lizard and paper. 

If there’s a tie (Both computer and user pick the same option), then the game ends in a draw.

**Comment by dovahkiin101** [comment]
My Analysis for this RPSLS game:

- There are two players in this game: The user and the computer.

- The game will start when the user (assuming that it is you) will pick one of 5 options (Rock, Paper, Scissor, Lizard, Spock). 

- Once clicked, the option he/she and the computer chooses will show up on the screen.

- Now, this is where we will determine which of the two players will get the points:

1. Rock:
Wins against: Scissor, Lizard
Loses against: Paper, Spock

2.Paper:
Wins against: Spock, Rock
Loses against: Lizard, Scissor

3.Scissors:
Wins against: Paper, Lizard
Loses against: Rock, Spock

4.Lizard:
Wins against: Paper, Spock
Loses against: Scissor, Rock

5.Spock:
Wins against:Scissors, Rock
Loses against:Lizard, Paper

Reference:
https://bigbangtheory.fandom.com/wiki/Rock,_Paper,_Scissors,_Lizard,_Spock

![RPSLS Rules](https://user-images.githubusercontent.com/84821316/120250598-b979c380-c24c-11eb-84e9-502c75f1b63a.png)

- This process will be repeated at least around 6 attempts.


- Now, when it comes to selecting the winners, if the user gains the most points than the computer, he/she wins. But vice versa if the computer wins.

- If both of the players gets equal points, then the game ends with a draw.

- Optional, but a "Try Again" Button can be placed within the game website, and when it's pressed, the game will restart once again.

DLA R.

**Comment by faustocorreaa** [comment]
Hi everyone this is my algorithm for RPSLS game:
![image](https://user-images.githubusercontent.com/84917984/120260610-d964b380-c25b-11eb-88c0-0100da93ccb5.png)

And this is the table to compare who wins:
![image](https://user-images.githubusercontent.com/84917984/120260676-f9947280-c25b-11eb-8c0e-a9e4a94ac39e.png)

**Comment by RodneyRamsay1** [comment]
Analysis for RPSLS game :

1. Start the game
2. The player is given multiple options (rock, paper, scissors, lizard, spock)
3. The player chooses one of the options
4. The computer chooses a random option
5. The player's choice is compared with the computers choice
6. Depending on the comparison, the player or the computer is declared the winner unless they both choose the same option which, will result in a draw.

Rules:
Scissors will cut paper. Paper will cover the rock. Rock will destroy the lizard. The lizard will poison spock. Spock will break the scissors. Scissors will destroy the lizard. Lizard will eat the paper. Paper beats spock. Spock beats the rock. Rock beats scissors.

**Comment by SyedNaufil** [comment]
Analysis of RPSLS game without technical language
1. start
2. options displayed to the player
3. player picks 1 of the options 
4. computer picks 1 of the choices at random which could include the same choice as the player's
5. both choices are compared
6. either computer could win , player could win or the outcome would be a draw
choices include 
scissors 
rock
paper
lizard
spock

**Comment by SyedNaufil** [comment]
This simple image shows the relationship between each of the options. we could give each object a property of "Strength" and "Weakness" according to this image. Depending on the option the player would lose if the computer picks a "Weakness" of the players choice and vice versa. 

Sample object could be something like this:

var rock = {strength: [scissors , lizard] ,weakness: [spock , paper]}

Source of image: https://steemit.com/programming/@dkmathstats/a-rock-paper-scissors-lizard-spock-chart-in-r 
![rock paper etc](https://user-images.githubusercontent.com/50566955/120347979-bfae8500-c2ca-11eb-9000-e1214cdd782f.png)

**Reply by ebajcar** [reply]
Good. The more alternatives you explore the better! 
Does this get into strategy how to win? This would be a great approach if one of the players is the machine.  In that case, the program would need to contain the logic how to win 👍

**Comment by afanasiev-kirill** [comment]
My analysis for RPSLS:
1. Start the game 
2.  Show options to user: rock, paper, scissors, lizard, spock.
2. User picks an option
3. Computer picks an option
4. Define who wins
5. Declare winner 
6. Suggest user playing more Yes/No, if user enter "Yes" - start game again, else end the game.

**Comment by mariomartz** [comment]
**My Analysis of RockPaperScissorsSpockLizard**

**Core Gameplay Loop**
1.	User is visually presented with 5 options.
      a.	Rock
      b.	Paper
      c.	Scissors
      d.	Spock
      e.	Lizard
2.	User is prompted to select from the options above.
3.	User’s choice is highlighted.
4.	Computer’s choice is revealed.
5.	Based on the user’s choice and the computer’s choice, the winner is revealed.
6.	User is then given the choice to play again or exit.

**Rules for Determining Winner**
![FIUAIWEI7Q0TCUT](https://user-images.githubusercontent.com/84930451/120264122-1e441680-c26b-11eb-86e3-7706aa56961e.jpg)

This image really helped me understand the rules of the game since it explains them in a visual manner.

Reference: https://www.instructables.com/How-to-Play-Rock-Paper-Scissors-Lizard-Spock/

**Comment by RyGil** [comment]
Analysis of the RPSLS game

General context:
-	The game expands upon the popular Rock, Paper, Scissors game
-	The rules of the game are the same as Rock, Paper, Scissors, with the addition of two more options a player can choose from

Rules:
The rules for the game are very straight forward.
-	Number of players required: two.
-	The players agree on the number of rounds a player needs to win in order to win the game (ie. best 2/3) – This is not mandatory.
-	Each player will choose one of the options available (Rock, Paper, Scissors, Lizard, or Spock).
-	After the count of three, both players use their hand to reveal their choice to the other player simultaneously.
-	If a tie occurs, players replay the round until the tie is broken. A tie can only occur when both players choose the same option.

What beats what?
-	Rock beats Scissors
-	Scissors cut Paper
-	Paper covers Rock
-	Lizard eats Paper
-	Rock crushes Lizard
-	Lizard poisons Spock
-	Scissors decapitates Lizard
-	Spock vaporizes Rock
-	Paper disproves Spock
-	Spock smashes Scissors

Visual representation of what beats what.
![image](https://user-images.githubusercontent.com/84523029/120270308-62d5af00-c277-11eb-94ac-33c7723c6ba6.png)

References: 
Rules: https://bigbangtheory.fandom.com/wiki/Rock,_Paper,_Scissors,_Lizard,_Spock
Image: https://www.amazon.ca/Rock-Paper-Scissors-Lizard-Spock/dp/B00C1ETH4I

**Comment by E-Annie** [comment]
My Analysis of RPSLS: 

1.	Start game by displaying 5 actions (rock, paper, scissors, lizard, spock) in pictures for the player to choose from
2.	Player chooses an action 
3.	Computer randomly generates an action 
4.	Both actions are displayed on the screen 
5.	The winner is chosen based on the actions chosen: 
 
a.	Case ROCK: 
o	Beats: scissors, lizard 
o	Loses: paper, spock 
b.	Case PAPER: 
o	Beats: rock, spock 
o	Loses: scissors, lizard 
c.	Case SCISSORS: 
o	Beats: paper, lizard 
o	Loses:  rock, spock 
d.	Case SPOCK: 
o	Beats: rock, scissors 
o	Loses: lizard, paper
e.	Case LIZARD: 
o	Beats:  paper, spock 
o	Loses: rock, scissors 

6.	Display the winner 
7.	If player press “Play Again!”, goes back to step 1.
8.	If player press “Home”, goes to member’s home screen

**Comment by Vrushabh7075** [comment]
var userChoice = prompt("Do you choose rock, paper, scissors, lizard, or spock?")
var computerChoice = Math.random();
if (computerChoice < 0.2) {
    computerChoice = "rock";
} else if (computerChoice <= 0.4) {
    computerChoice = "paper";
} else if (computerChoice <= 0.6) {
    computerChoice = "scissors";
} else if (computerChoice <= 0.8) {
    computerChoice = "lizard";
} else {
    computerChoice = "spock";
}

alert("The computer chose " + computerChoice);

var compare = function(choice1, choice2){
    if (choice1 === choice2) {
        alert("And... It's a tie!");
    }

//If the user chose rock...
else if (choice1 === "rock") {
    if (choice2 === "scissors") {
        alert("Rock wins!");
    } else if (choice2 === "paper") {
        alert("Paper wins!");
    } else if (choice2 === "lizard") {
        alert("Rock wins!");
    } else {
        alert("Spock wins!");
    }
}

//If the user chose paper...
else if (choice1 === "paper") {
    if (choice2 === "scissors") {
        alert("Scissors wins!");
    } else if (choice2 === "rock") {
        alert("Paper wins!");
    } else if (choice2 === "lizard") {
        alert("Lizard wins!");
    } else {
        alert("Paper wins!");
    }
}

//If the user chose scissors...
else if (choice1 === "scissors") {
    if (choice2 === "paper") {
        alert("Scissors wins!");
    } else if (choice2 === "rock") {
        alert("Rock wins!");
    } else if (choice2 === "lizard") {
        alert("Scissors wins!");
    } else {
        alert("Spock wins!");
    }
}

//If the user chose lizard...
else if (choice1 === "lizard") {
    if (choice2 === "scissors") {
        alert("Scissors wins!");
    } else if (choice2 === "rock") {
        alert("Rock wins!");
    } else if (choice2 === "paper") {
        alert("Lizard wins!");
    } else {
        alert("Lizard wins!");
    }
}

//If the user chose spock...
else if (choice1 === "spock") {
    if (choice2 === "scissors") {
        alert("Spock wins!");
    } else if (choice2 === "rock") {
        alert("Spock wins!");
    } else if (choice2 === "lizard") {
        alert("Lizard wins!");
    } else {
        alert("Paper wins!");
    }
}
};
compare(userChoice, computerChoice);




there is a problem In this code..if user inputs it doesn't recognize as valid inputs

**Reply by kmoran905** [reply]
The rules of the game.

Scissors > Paper
Scissors > Lizard
Rock > Scissors
Rock > Lizard
Lizard > Paper
Lizard > Spock
Spock > Rock
Spock > Scissors
Paper > Spock
Paper > Rock

References: https://bigbangtheory.fandom.com/wiki/Rock,_Paper,_Scissors,_Lizard,_Spock

**Reply by ebajcar** [reply]
It is not clear what the "rules of the game" are from the description. What does the symbol > represent? 
It is important to document using full sentences. Keep in mind that documentation is meant to read by other people.

**Reply by ebajcar** [reply]
@Vrushabh7075 I am not sure what you are trying to do.  
If this is an actual implementation, please make sure you use unobtrusive Javascript, specifically avoid using prompt and alert control boxes (study the class material and examples further).  
If it is meant to be an algorithm, algorithms are technology agnostic, in other words, they do not contain references to specific languages, they are meant to describe the step-by-step process in human-readable form (plain English).  Please refer to class material as this was covered.

**Comment by ferkopa** [comment]
Hello Professor.
Here is my analysis of the RPSLS game.
-	It is based on the traditional game of Rock, Paper and, Scissors, where players select one of the 3 choices and battle against 1 or more players.
-	However, when the show “Big Bang Theory” introduced two new players into the game, it changed the meaning of the game.
Analysis of the game:
1.	User clicks on RPSLS from the main page
2.	User clicks the welcome button, prompting the game to begin
3.	User picks any of the 5 images, rock, paper, scissors, lizard, or Spock
4.	Upon choosing, the computer will generate a random image to battle the user’s pick
5.	The following will determine who wins, who loses, or if it is a draw
     a.    Scissors beats paper
     b.    Paper beats rock
     c.	    Rock beats lizard
     d.    Lizard beats Spock
     e.    Spock beats scissors
     f.	    Scissors beats lizard
     g.    Lizard beats paper
     h.    Paper beats Spock
     i.	    Spock beats rock
     j.	    Rock beats scissors

6.	The user will be notified of the results
7.	If the user wishes to play again, they may click the play again button in navbar
8.	If they wish to play another game, they may click the home button and return to the logged-in member webpage

Reference: https://bigbangtheory.fandom.com/wiki/Rock,_Paper,_Scissors,_Lizard,_Spock

**Comment by OmkarKhurana** [comment]
Hello Professor.
Here is my analysis of the RPSLS game.
This is a game based on the classic game The Rock Paper Scissor. 
The show Big Bang Theory added two new elements known as the Lizard and the Spock, to increase the interest of the newer generation in the game and also to make it more interesting.

The Rules of the games are: 
-Number of players required: two.
-Each player will choose one of the elements available (Rock, Paper, Scissors, Lizard, or Spock).
-After the count of three, both players use their hands to reveal their choice to the other player simultaneously.
-A tie only occurs when both players choose the same element.
-If a tie occurs, players replay until the tie is broken. 
-If a tie is not the result then the score is calculated(optional) after multiple rounds of the game.

The Flow of the game(Website):
1. User selects RPSLS from the member's homepage
2. They are presented by 5 images of the 5 main elements: ROCK, PAPER, SCISSORS, LIZARD and SPOCK.
3. User clicks on one of the images selecting them.
4. The computer selects an element based on a randomly generated number( which will be further used in a switch case to assign the element.).
5. Then the decision is made using the following logic:
    a. ROCK:
            Beats LIZARD AND SCISSORS
            Loses from Spock PAER AND SPOCK
    b. Case PAPER:
            Beats ROCK AND SPOCK
            Loses from SCISSORS AND LIZARD
    c. Case SCISSORS:
            Beats PAPER AND LIZARD
            Loses from ROCK AND SPOCK
    d. Case SPOCK:
            Beats ROCK AND SCISSORS
            Loses  from LIZARD AND PAPER
    e. Case LIZARD:
            Beats PAPER AND SPOCK
            Loses from ROCK AND SCISSORS
6. The result is then displayed and the user can play as many times as he/she wants.
7. The score is calculated.
8. -ve score means user loses/ +ve score means the user wins.
9.  User can reset to start the game with a zero score.

**Comment by nusrathsyed** [comment]
Analysis for RPSLS:

1. The user clicks on START GAME button.
2. The user clicks on 1 symbol out of 5 -- > Rock, Paper, Lizard, Scissors, or Spock
3. The computer randomly picks one of the symbols after.
4. Show symbol user clicked and show symbol computer picked.
5. Decide who is the winner using statements in game:
 - rock beats scissors or lizard
 - paper beats rock or spock
 - scissors beats lizard or paper
 - lizard beats paper or spock
 - spock beats rock or scissors
6. Play again button will open a new game. 
7. END GAME button will end the game.
